pkg/imrpc/proxy.proto

```diff
@@ -0,0 +1,309 @@
+syntax="proto3";
+
+package imrpc;
+
+import "google/protobuf/empty.proto";
+import "google/protobuf/timestamp.proto";
+import "github.com/longhorn/longhorn-engine/proto/ptypes/controller.proto";
+import "github.com/longhorn/longhorn-engine/proto/ptypes/syncagent.proto";
+
+service ProxyEngineService {
+	rpc Ping(google.protobuf.Empty) returns (google.protobuf.Empty);
+
+	rpc ServerVersionGet(ProxyEngineRequest) returns (EngineVersionProxyResponse);
+
+	rpc VolumeGet(ProxyEngineRequest) returns (EngineVolumeGetProxyResponse);
+	rpc VolumeExpand(EngineVolumeExpandRequest) returns (google.protobuf.Empty);
+	rpc VolumeFrontendStart(EngineVolumeFrontendStartRequest) returns (google.protobuf.Empty);
+	rpc VolumeFrontendShutdown(ProxyEngineRequest) returns (google.protobuf.Empty);
+
+	rpc VolumeSnapshot(EngineVolumeSnapshotRequest) returns (EngineVolumeSnapshotProxyResponse);
+	rpc SnapshotList(ProxyEngineRequest) returns (EngineSnapshotListProxyResponse);
+	rpc SnapshotRevert(EngineSnapshotRevertRequest) returns (google.protobuf.Empty);
+	rpc SnapshotPurge(EngineSnapshotPurgeRequest) returns (google.protobuf.Empty);
+	rpc SnapshotPurgeStatus(ProxyEngineRequest) returns (EngineSnapshotPurgeStatusProxyResponse);
+	rpc SnapshotClone(EngineSnapshotCloneRequest) returns (google.protobuf.Empty);
+	rpc SnapshotCloneStatus(ProxyEngineRequest) returns (EngineSnapshotCloneStatusProxyResponse);
+	rpc SnapshotRemove(EngineSnapshotRemoveRequest) returns (google.protobuf.Empty);
+
+	rpc SnapshotBackup(EngineSnapshotBackupRequest) returns (EngineSnapshotBackupProxyResponse);
+	rpc SnapshotBackupStatus(EngineSnapshotBackupStatusRequest) returns (EngineSnapshotBackupStatusProxyResponse);
+	rpc BackupRestore(EngineBackupRestoreRequest) returns (EngineBackupRestoreProxyResponse);
+	rpc BackupRestoreStatus(ProxyEngineRequest) returns (EngineBackupRestoreStatusProxyResponse);
+	rpc BackupVolumeList(EngineBackupVolumeListRequest) returns (EngineBackupVolumeListProxyResponse);
+	rpc BackupVolumeGet(EngineBackupVolumeGetRequest) returns (EngineBackupVolumeGetProxyResponse);
+	rpc BackupGet(EngineBackupGetRequest) returns (EngineBackupGetProxyResponse);
+	rpc BackupConfigMetaGet(EngineBackupConfigMetaGetRequest) returns (EngineBackupConfigMetaGetProxyResponse);
+	rpc BackupRemove(EngineBackupRemoveRequest) returns (google.protobuf.Empty);
+
+	rpc ReplicaAdd(EngineReplicaAddRequest) returns (google.protobuf.Empty);
+	rpc ReplicaList(ProxyEngineRequest) returns (EngineReplicaListProxyResponse);
+	rpc ReplicaRebuildingStatus(ProxyEngineRequest) returns (EngineReplicaRebuildStatusProxyResponse);
+	rpc ReplicaVerifyRebuild(EngineReplicaVerifyRebuildRequest) returns (google.protobuf.Empty);
+	rpc ReplicaRemove(EngineReplicaRemoveRequest) returns (google.protobuf.Empty);
+}
+
+message ProxyEngineRequest{
+	string address = 1;
+}
+
+message EngineVersionProxyResponse {
+	ptypes.VersionOutput version = 2;
```

Comment:
hmmmm... Not sure why the number starts from 2. Similar below

Reply:
Fixed.